src/Admin/AbstractAdmin.php

```diff
@@ -952,6 +952,13 @@ public function getParentAssociationMapping()
      */
     final public function addParentAssociationMapping($code, $value)
     {
+        if (\is_string($this->parentAssociationMapping)) {
+            @trigger_error(sprintf(
+                'Calling "%s"  when $this->parentAssociationMapping is string is deprecated and will be removed in 4.0.',
```

Comment:
```suggestion
                'Calling "%s"  when $this->parentAssociationMapping is string is deprecated since sonata-admin/admin-bundle 3.x and will be removed in 4.0.',
```

But, do we need this deprecation ?
```
$this->parentAssociationMapping[$code] = $value;
```
will fail if it's a string, no ?

Reply:
@VincentLanglet it depends on the type of `$code`

this will fail http://sandbox.onlinephpfunctions.com/code/1131c95a27e07e145bc849eec493e7ab862ad8b1
this will not http://sandbox.onlinephpfunctions.com/code/ecbeef93d8d90a4e50ed4300b1ddfd8f052b7ac3


-----
sanding with http://sandbox.onlinephpfunctions.com because 3v4l.org/ doesn't work at the moment